reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2011-Present VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import org.reactivestreams.Subscriber;
+
+import reactor.util.annotation.Nullable;
+import reactor.util.concurrent.Queues;
+import reactor.util.context.Context;
+
+/**
+ * A collection of standalone sinks ({@link StandaloneFluxSink} and
+ * {@link StandaloneMonoSink}).
+ *
+ * @author Simon Baslé
+ */
+public final class Sinks {
+
+	private Sinks() { }
+
+	/**
+	 * A {@link StandaloneFluxSink} with the following characteristics:
+	 * <ul>
+	 *     <li>Multicast</li>
+	 *     <li>Backpressure : this sink honors downstream demand by conforming to the lowest demand in case
+	 *     of multiple subscribers.</li>
+	 *     <li>Replaying: No replay. Only forwards to a {@link Subscriber} the elements that have been
+	 *     pushed to the sink AFTER this subscriber was subscribed. To the exception of the first
+	 *     subscriber (see below).</li>
+	 *     <li>Without {@link Subscriber}: warm up. Remembers up to {@link Queues#SMALL_BUFFER_SIZE}
+	 *     elements pushed before the first {@link Subscriber} is registered.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> multicast() {
+		return new FluxProcessorSink<>(EmitterProcessor.create(Queues.SMALL_BUFFER_SIZE));
+	}
+
+	/**
+	 * A {@link StandaloneFluxSink} with the following characteristics:
+	 * <ul>
+	 *     <li>Multicast</li>
+	 *     <li>Backpressure : this sink honors downstream demand of individual subscribers.</li>
+	 *     <li>Replaying: No replay. Only forwards to a {@link Subscriber} the elements that have been
+	 *     pushed to the sink AFTER this subscriber was subscribed.</li>
+	 *     <li>Without {@link Subscriber}: Discarding. Pushing elements while there are no {@link Subscriber}
+	 *     registered will simply discard these elements instead of "warming up" the sink.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> multicastNoWarmup() {
```

Comment:
"warmup" sounds like a new term added. And, TBH, I struggle to understand it (even after reading the javadoc) 😅 
Could you please help me understanding the concept?

Reply:
yeah I'm having trouble finding a better term. It's the way some processors will "remember" upstream `onNext` (or in our case `sink.next`) that come in BEFORE the first `Subscriber` subscribes. But then the second `Subscriber` would only see live pushes.

```
\ denotes a subscription
<-warmup------><---------hot ---------->

--(1)---(2)----\---------\----(3)------>   sink
              -\(1)(2)--------(3)------>   sub1
                        -\----(3)------>   sub2
```

Reply:
Thanks, the diagram really helps! Maybe let's add it to the javadoc?

Reply:
I'll add a quick SVG version for both `multicast()` and `multicastNoWarmup()`